backend/src/main/java/com/huellapositiva/domain/actions/JoinProposalAction.java

```diff
@@ -0,0 +1,23 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.domain.service.ProposalService;
+import com.huellapositiva.domain.service.VolunteerService;
+import com.huellapositiva.infrastructure.orm.model.Volunteer;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class JoinProposalAction {
+
+    private final ProposalService proposalService;
+
+    private final VolunteerService volunteerService;
+
+    public void execute(Integer proposalId){
+        String volunteerEmail = (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
Retrieving the volunteerEmail could be done at controller level and passing it as a parameter so that we decouple this action's business logic from Spring. It would be a better OO approach with many benefits including loose decoupling, testability, maintainability, separation of concerns, ...